lib/Doctrine/ORM/Internal/Hydration/ObjectHydrator.php

```diff
@@ -148,6 +151,12 @@ protected function hydrateAllData()
             $coll->takeSnapshot();
         }
 
+        foreach ($this->uninitializedCollections as $coll) {
```

Comment:
I believe you need to reset this in cleanup()

Reply:
Indeed, makes sense

Reply:
Shouldn't there be a test for the cleanup of the new property then?

Reply:
I believe that the details of the realization are not something that needs to be tested by itself. There shouldn't be tests that break for on a correctly working class with an unchanged interface. Also, there are no tests for other collections or private fields that also getting reset, for example.

Reply:
Tests for checking if collections are getting initialized or not in different circumstances are nice to have but those are not related to the original issue.